controllers/infra/capability/configmap/configmap_capability_controller.go

```diff
@@ -71,70 +72,57 @@ func AddToManager(ctx *pkgctx.ControllerManagerContext, mgr manager.Manager) err
 		&handler.TypedEnqueueRequestForObject[*corev1.ConfigMap]{},
 		predicate.TypedFuncs[*corev1.ConfigMap]{
 			CreateFunc: func(e event.TypedCreateEvent[*corev1.ConfigMap]) bool {
-				return e.Object.Name == capabilities.WCPClusterCapabilitiesConfigMapObjKey.Name
+				return e.Object.Name == capabilities.ConfigMapKey.Name
 			},
 			UpdateFunc: func(e event.TypedUpdateEvent[*corev1.ConfigMap]) bool {
-				return e.ObjectOld.Name == capabilities.WCPClusterCapabilitiesConfigMapObjKey.Name
+				return e.ObjectOld.Name == capabilities.ConfigMapKey.Name
 			},
 			DeleteFunc: func(e event.TypedDeleteEvent[*corev1.ConfigMap]) bool {
 				return false
 			},
-			GenericFunc: func(e event.TypedGenericEvent[*corev1.ConfigMap]) bool {
-				return false
-			},
 		},
 		kubeutil.TypedResourceVersionChangedPredicate[*corev1.ConfigMap]{},
 	))
 }
 
 func NewReconciler(
 	ctx context.Context,
-	cache client.Reader,
+	client ctrlclient.Reader,
 	logger logr.Logger,
 	recorder record.Recorder) *Reconciler {
 
 	return &Reconciler{
 		Context:  ctx,
-		Cache:    cache,
+		Client:   client,
 		Logger:   logger,
 		Recorder: recorder,
 	}
 }
 
 type Reconciler struct {
 	Context  context.Context
-	Cache    client.Reader
+	Client   ctrlclient.Reader
 	Logger   logr.Logger
 	Recorder record.Recorder
 }
 
 // +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch
 
-func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	ctx = pkgcfg.JoinContext(ctx, r.Context)
-
-	if req.NamespacedName == capabilities.WCPClusterCapabilitiesConfigMapObjKey {
-		return ctrl.Result{}, r.reconcileWcpClusterCapabilitiesConfig(ctx, req)
-	}
-
-	r.Logger.Error(nil, "Reconciling unexpected object", "req", req.NamespacedName)
-	return ctrl.Result{}, nil
-}
+func (r *Reconciler) Reconcile(
+	ctx context.Context,
+	req ctrl.Request) (ctrl.Result, error) {
 
-func (r *Reconciler) reconcileWcpClusterCapabilitiesConfig(ctx context.Context, req ctrl.Request) error {
-	oldFeatures := pkgcfg.FromContext(ctx).Features
-	r.Logger.Info("Reconciling WCP Cluster Capabilities Config", "configMap", req.NamespacedName,
-		"isAsyncSVUpgrade", oldFeatures.SVAsyncUpgrade)
+	ctx = pkgcfg.JoinContext(ctx, r.Context)
 
-	cm := &corev1.ConfigMap{}
-	if err := r.Cache.Get(ctx, req.NamespacedName, cm); err != nil {
-		return client.IgnoreNotFound(err)
+	var obj corev1.ConfigMap
+	if err := r.Client.Get(ctx, req.NamespacedName, &obj); err != nil {
+		return ctrl.Result{}, err
```

Comment:
`ctrlclient.IgnoreNotFound(err)` (and in the other controller)? We're off the rails if it gets deleted but no point in forcing an error retry.